pkg/plugins/pg/source/snapshot/snapshot_integration_test.go

```diff
@@ -64,24 +72,55 @@ func TestPrematureDBClose(t *testing.T) {
 	s, err := NewSnapshotter(db, "records", []string{"id",
 		"column1", "key"}, "key")
 	assert.Ok(t, err)
-	// assert that we have at least one row and it's loading as expected
 	next1 := s.HasNext()
 	assert.Equal(t, true, next1)
-	// teardown to prematurely kill our DB connection and assert we get
-	// an ErrSnapshotInterrupt error
 	teardownErr := s.Teardown()
 	assert.True(t, cerrors.Is(teardownErr, ErrSnapshotInterrupt),
 		"failed to get snapshot interrupt error")
-	// assert Next fails because we have no rows to read.
-	_, err = s.Next()
+	_, err = s.Next(context.Background())
 	assert.Error(t, err)
-	// assert calling HasNext again returns false.
 	next2 := s.HasNext()
 	assert.Equal(t, false, next2)
-	// finally , assert calling Next when HasNext has returned false returns
-	// an ErrNoRows error for calling after database is closed.
-	rec, err := s.Next()
-	assert.Equal(t, rec, record.Record{})
+	rec, err := s.Next(context.Background())
+	assert.Equal(t, rec, sdk.Record{})
 	assert.True(t, cerrors.Is(err, ErrNoRows),
 		"failed to get snapshot incomplete")
 }
+
+// getTestPostgres is a testing helper that fails if it can't setup a Postgres
+// connection and returns a DB and the connection string.
+// * It starts and migrates a db with 5 rows for Test_Read* and Test_Open*
+func getTestPostgres(t *testing.T) *sql.DB {
+	prepareDB := []string{
+		// drop any existing data
+		`DROP TABLE IF EXISTS records;`,
+		// setup records table
+		`CREATE TABLE IF NOT EXISTS records (
+		id bigserial PRIMARY KEY,
+		key bytea,
+		column1 varchar(256),
+		column2 integer,
+		column3 boolean);`,
+		// seed values
+		`INSERT INTO records(key, column1, column2, column3)
+		VALUES('1', 'foo', 123, false),
+		('2', 'bar', 456, true),
+		('3', 'baz', 789, false),
+		('4', null, null, null);`,
+	}
```

Comment:
Unrelated to this PR, but I'm wondering if we can add columns of other data types (e.g. geo types, array types, timestamps etc.), so that we have more extensive tests?

Reply:
Yes, I agree, we should test JSON and JSONB, geo types, etc... as well. I have some other test cases in git stash that I'm going to make into PRs.